debian/stackdriver-agent.init

```diff
@@ -74,115 +72,35 @@ if [ -n "$JAVA_LIB_DIR" ]; then
-#     AUTOGENERATE_COLLECTD_CONFIG=\"no\"
-# in /etc/default/$NAME
-"
-    if [ -z "$USE_GCM_API" -a -n "$DETECT_GCM" ]; then
-        # automatically determine if we should be using the GCM API
-        # first, see if the instance has a monitoring.write scope
-        INSTANCE_SCOPES=$(curl --silent -f -H "Metadata-Flavor: Google" http://169.254.169.254/computeMetadata/v1/instance/service-accounts/default/scopes 2>/dev/null)$'\n'
-        if [[ $INSTANCE_SCOPES == *"/monitoring.write"$'\n'* || $INSTANCE_SCOPES == *"/monitoring"$'\n'* || $INSTANCE_SCOPES == *"/cloud-platform"$'\n'* ]]; then
-            log_progress_msg "Found sufficient authorization scope for the GCM API"
-            USE_GCM_API="yes"
-        # second, see if the application default credential file is in the system location
-        elif [ -f "${GOOGLE_APPLICATION_CREDENTIALS:-/etc/google/auth/application_default_credentials.json}" ]; then
-            log_progress_msg "Found a credentials file"
-            USE_GCM_API="yes"
-        else
-            log_warning_msg "Not authorized to talk to the GCM API, falling back on legacy"
-            USE_GCM_API="no"
-        fi
-    fi
-    # now see if we should be using write_gcm and use the appropriate template if so
-    if [ "$USE_GCM_API" = "yes" ]; then
-        echo "$AUTOGENERATE_HEADER" > $CONFIGFILE
-        cat $GCM_CONFIG_TEMPLATE >> $CONFIGFILE
-        return
-    fi
-
-    if test -z "$STACKDRIVER_API_KEY" ; then
-        STACKDRIVER_API_KEY=$(curl --silent -f -H "Metadata-Flavor: Google" http://169.254.169.254/computeMetadata/v1/project/attributes/stackdriver-agent-key 2>/dev/null)
-        if test -z "$STACKDRIVER_API_KEY"; then
-            log_failure_msg "Need to set STACKDRIVER_API_KEY in /etc/default/$NAME"
-            return 3
-        fi
-    fi
-
-    local IID=$(get_instance_id)
-    if [ -z "$IID" ]; then
-        log_failure_msg "Unable to discover instance id"
-        return
-    fi
-    if [ -z "$COLLECTD_ENDPOINT"]; then
-        COLLECTD_ENDPOINT=$(get_collectd_endpoint)
-        if [[ $? -ne 0 ]]; then
-            return 1
-        fi
-    fi
-    echo "$AUTOGENERATE_HEADER" > $CONFIGFILE
-    sed -e "s/{IID}/$IID/; s/{API_KEY}/$STACKDRIVER_API_KEY/; s|{COLLECTD_ENDPOINT}|$COLLECTD_ENDPOINT|" $CONFIG_TEMPLATE >> $CONFIGFILE
-}
-
 # return:
 #   0 if config is fine
 #   1 if there is a syntax error
 #   2 if there is no configuration
+#   3 if the instance doesn't have the necessary credentials
 check_config() {
-    if test ! -e "$CONFIGFILE"; then
+    if test ! -e "$CONFIG"; then
         return 2
     fi
-    if ! $DAEMON -t -C "$CONFIGFILE"; then
+    if ! $DAEMON -t -C "$CONFIG"; then
         return 1
     fi
+    # Check if the application default credentials file is in the system
+    # location.
+    if [ ! -f "${GOOGLE_APPLICATION_CREDENTIALS:-/etc/google/auth/application_default_credentials.json}" ]; then
```

Comment:
Looks like you're also picking up #27...

Reply:
Oh yeah. I'll add that in the description